packages/transform-to-vanilla/src/transform-values/merge-values.ts

```diff
@@ -0,0 +1,189 @@
+export function mergeToComma(values: MergeNeedValues): string;
+export function mergeToComma(values: MergeNeedFallbackValues): FallbackValues;
+export function mergeToComma(values: MergeNeedComplexValues): MergedValues {
+  const separator = ", ";
+  if (isSingleArray(values)) {
+    return values.join(separator);
+  }
+
+  return transformArray(values, separator);
+}
+
+export function mergeToSpace(values: MergeNeedValues): string;
+export function mergeToSpace(values: MergeNeedFallbackValues): FallbackValues;
+export function mergeToSpace(values: MergeNeedComplexValues): MergedValues {
+  const separator = " ";
+  if (isSingleArray(values)) {
+    return values.join(separator);
+  }
+
+  return transformArray(values, separator);
+}
+
+// == Utils ====================================================================
+function isSingleArray(
+  values: MergeNeedComplexValues
+): values is MergeNeedValues {
+  return Array.isArray(values) && !values.some(Array.isArray);
+}
+
+interface MergeStackElement {
+  index: number;
+  string: string;
+}
+function transformArray(
+  values: MergeNeedFallbackValues,
+  joinMethod: ", " | " " = ", "
+) {
+  const result: FallbackValues = [];
+  const stack: MergeStackElement[] = [{ index: 0, string: "" }];
+  const valueLength = values.length;
+
+  while (stack.length > 0) {
+    const { index, string } = stack.pop() as MergeStackElement;
+
+    if (index >= valueLength) {
+      result.push(string);
+      continue;
+    }
+
+    const currentElement = values[index];
+    const nextIndex = index + 1;
+    const seperatedStr = `${string}${index === 0 ? "" : joinMethod}`;
+    if (Array.isArray(currentElement)) {
+      for (let i = currentElement.length - 1; i >= 0; i--) {
+        stack.push({
+          index: nextIndex,
+          string: seperatedStr + currentElement[i]
+        });
+      }
+    } else {
+      stack.push({
+        index: nextIndex,
+        string: seperatedStr + currentElement
+      });
+    }
+  }
+
+  return result;
+}
+
+// == Tests ====================================================================
+if (import.meta.vitest) {
+  const { describe, it, expect, expectTypeOf } = import.meta.vitest;
+
+  describe.concurrent("mergeToComma", () => {
+    it("Merge with Single Array", () => {
+      const sample = ["inset 0 0 10px #555", "0 0 20px black"];
+      const result = mergeToComma(sample);
+
+      expect(result).toBe("inset 0 0 10px #555, 0 0 20px black");
+      expectTypeOf(result).toBeString();
+    });
+
+    it("Merge with a Multi-Values Fallback Array", () => {
+      const sample = [
+        "inset 0 0 10px #555",
+        ["0 0 20px black", "0 0 20px rgba(0, 255, 0, 0.5)"]
+      ];
+      const result = mergeToComma(sample);
+
+      expect(result).toStrictEqual([
+        "inset 0 0 10px #555, 0 0 20px black",
+        "inset 0 0 10px #555, 0 0 20px rgba(0, 255, 0, 0.5)"
+      ]);
+      expectTypeOf(result).toEqualTypeOf<string[]>();
```

Comment:
Fallback array will be return `FallbackValues` at funtion signiture.
```typescript
export function mergeToComma(values: MergeNeedValues): string;
export function mergeToComma(values: MergeNeedFallbackValues): FallbackValues;
export function mergeToComma(values: MergeNeedComplexValues): MergedValues {
}
```

IMO, t would be better to use `FallbackValues` instead of `string[]` to make the meaning clear.

Reply:
thx to check, i changed that now